src/lakefs_spec/spec.py

```diff
@@ -441,31 +447,95 @@ def _open(
             **kwargs,
         )
 
+    def put_file_to_blockstore(
+        self, lpath, repository, branch, resource, presign=True, storage_options={}
+    ):
+        print("In Blockstore")
+        if tuple(int(v) for v in __lakefs_client_version__.split(".")) < (0, 111, 0):
+            blockstore_type = self.client.config_api.get_storage_config().blockstore_type
+        else:
+            print("OOOKS")
+            print(self.client.config_api.get_config())
+            print(self.client.config_api.get_config().storage_config)
+            blockstore_type = self.client.config_api.get_config().storage_config.blockstore_type
+            print(blockstore_type)
+        if blockstore_type == "local":
+            raise ValueError(
+                "Cannot write to blockstore of type 'local'. Disable use_blockstore or configure remote blockstore."
+            )
+
+        staging_location = self.client.staging_api.get_physical_address(
+            repository, branch, resource, presign=presign
+        )
+
+        if presign:
+            remote_url = staging_location.presigned_url
+            content_type, _ = mimetypes.guess_type(lpath)
+            if content_type is None:
+                content_type = "application/octet-stream"
+            with open(lpath, "rb") as f:
+                headers = {
+                    "Content-Type": content_type,
+                }
+                request = urllib.request.Request(
+                    url=remote_url, data=f, headers=headers, method="PUT"
+                )
+                try:
+                    if not remote_url.lower().startswith("http"):
+                        raise ValueError("Wrong protocol for remote connection")
+                    else:
+                        with urllib.request.urlopen(
+                            request
+                        ):  # nosec [B310:blacklist] # We catch faulty protocols above.
+                            logger.info(f"Successfully uploaded: {lpath} ")
```

Comment:
Maybe add a log statement before the actual upload as well, so the user sees something in case of slow connections